src/encoded/static/components/search.js

```diff
@@ -735,6 +735,289 @@ TypeTerm.propTypes = {
+                // Check if now the years match and month lists need to be limited
+                this.checkForSameYear();
+            });
+            // Possibilities for endYears must now all be greater than the new startYear
+            const endYears = this.state.possibleYears.filter(year => +year >= event.target.value);
+            this.setState({ endYears });
+        // We are changing the end year, which means we need to change the possiblities for the starting year and also the possible end months
+        } else {
+            // Set endYear to be user choice
+            this.setState({ endYear: event.target.value }, () => {
+                // Check if now the years match and month lists need to be limited
+                this.checkForSameYear();
+            });
+            // Possiblities for startYears must now all be less than the new endYears
+            const startYears = this.state.possibleYears.filter(year => +year <= event.target.value);
+            this.setState({ startYears });
+        }
+    }
+
+    resetMonthDropDowns() {
+        this.setState({ startMonths: this.state.possibleMonths });
+        this.setState({ startMonth: '01' });
+        this.setState({ endMonths: this.state.possibleMonths });
+        this.setState({ endMonth: '12' });
+    }
+
+    // If the start year and the end year match, we have to be careful to not allow the user to pick an end month that is earlier than the start month
+    checkForSameYear() {
+        if (this.state.startYear === this.state.endYear) {
+            // If start month is later than the end month and years match, this is not allowed, so we reset
+            if (this.state.endMonth < this.state.startMonth) {
+                this.resetMonthDropDowns();
+            // If start and end months are allowed, we still need to filter dropdown possible lists so they can't select an unallowed combination
+            } else {
+                // endMonths can only display months that are after the chosen startMonth
+                const endMonths = this.state.possibleMonths.filter(month => +month >= +this.state.startMonth);
+                this.setState({ endMonths });
+                // startMonths can only display months that are before the chosen endMonth
+                const startMonths = this.state.possibleMonths.filter(month => +month <= +this.state.endMonth);
+                this.setState({ startMonths });
+            }
+        // If the start and end years previously matched (but now they don't), an incomplete list of months may be set and we need to update
+        } else {
+            if (this.state.possibleMonths !== this.state.startMonths) {
+                this.setState({ startMonths: this.state.possibleMonths });
+            }
+            if (this.state.possibleMonths !== this.state.endMonths) {
+                this.setState({ endMonths: this.state.possibleMonths });
+            }
```

Comment:
These are array comparisons — seeing if possibleMonths is the same _instance_ of array as startMonths and endMonths, which we can't really control since these states belong to React. Do you mean if these arrays have the same _contents_? You’d need a different test (I think there’s an ES6 or ES7 shortcut to see if arrays have the same contents; not sure).

Reply:
I think it's ok to just compare the lengths of the arrays because if they're the same length, their contents should match. I did mean to compare the contents, though.